2018/004-image-update-caasp-v4.md

```diff
@@ -0,0 +1,660 @@
+without introducing technical debt, without making the whole upgrade story worse
+than the one of v3 and without having to write that much code.
+
+V3 already has some limits, we are fine keeping them around for 4.0:
+
+  * Nodes have to be rebooted to get latest images.
+  * Admin nodes images are always using the latest version available on the
+    node.
+  * Nodes are rebooted even when not actually needed: images are delivered via
+    RPMs, when a new velum image is published all the nodes of the cluster are
+    going to be rebooted even if the only one that actually needs this package
+    is the admin node.
+  * Kubernetes add-ons (dex, tiller, kubedns) are currently referenced by their
+    short tag.
+  * Kubernetes daemon-sets (flannel) are referenced by their short tag.
+  * Static pods on the Kubernetes nodes (haproxy) are referenced by their
+    short tag.
+  * The short tag consists only of the version of the core application shipped
+    inside of the image. If the image is rebuilt due to a bug or a security fix
+    the short tag won't change. The usage of short tags can lead customers to
+    run unpatched images or bugged ones without being aware of that.
+
+The proposed solution for 4.0 is going to keep some of these limitations (reboots)
+but is going to improve the security aspects.
+
+## Detailed RFC
+
+> In this section of the document the target audience is the dev team. Upon
+> reading this section each engineer should have a rather clear picture of what
+> needs to be done in order to implement the described feature. 
+
+### Changes for image building
+
+Images are going to have 3 tags:
+
+  * **short:** this is the core application version. For example, given `2.6.2`
+    is the version of the `docker-registry` application, the short tag of its
+    image is going to be `2.6.2`.
+  * **intermediate:** this is core application version plus the image revision
+    number. The revision number is written into the KIWI spec file, it's updated
+    manually by the developers every time a fix is done to the image.
+    The intermediate tag for the first iteration of docker-registry image
+    would be `2.6.2-rev1`.
+  * **long:** intermediate tag + build numbers generated by the Open Build
+    Service. The long tag for the first iteration of docker-registry image
+    would be something like `2.6.2-rev1+5.1`, where `5.1` are numbers that
+    are automatically incremented by the Open Build Service every time the
+    image is built.
+
+Images are going to be pushed to the SUSE registry, they are not going to be
+wrapped into RPMs anymore.
+
+We will keep using a post-build hook on the build service to create an RPM file
+per image: `<container-image-name>-metadata`. This RPM file will have a simple
+text file inside of it with the metadata of the image (like all its tags).
+
+The contents of the file are not actually useful, the purpose of this RPM is
```

Comment:
This mechanism won't work at all or not well - releasing containers and RPM packages is completely independent.
Even if the container release mechanism was able to handle RPMs as well, it would not be synchronized with the containers currently. That won't be easy to do.